beacon-chain/rpc/eth/light-client/handlers.go

```diff
@@ -0,0 +1,402 @@
+	ctx, span := trace.StartSpan(req.Context(), "beacon.GetLightClientBootstrap")
+	defer span.End()
+
+	// Get the block
+	blockRootParam, err := hexutil.Decode(mux.Vars(req)["block_root"])
+	if err != nil {
+		http2.HandleError(w, "invalid block root "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var blockRoot [32]byte
+	copy(blockRoot[:], blockRootParam)
+	blk, err := bs.BeaconDB.Block(ctx, blockRoot)
+	if !shared.WriteBlockFetchError(w, blk, err) {
+		return
+	}
+
+	// Get the state
+	state, err := bs.Stater.StateBySlot(ctx, blk.Block().Slot())
+	if err != nil {
+		http2.HandleError(w, "could not get state "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	bootstrap, err := CreateLightClientBootstrap(ctx, state)
+	if err != nil {
+		http2.HandleError(w, "could not get light client bootstrap "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	response := &LightClientBootstrapResponse{
+		Version: ethpbv2.Version(blk.Version()).String(),
+		Data:    bootstrap,
+	}
+
+	http2.WriteJson(w, response)
+}
+
+// GetLightClientUpdatesByRange - implements https://github.com/ethereum/beacon-APIs/blob/263f4ed6c263c967f13279c7a9f5629b51c5fc55/apis/beacon/light_client/updates.yaml
+func (bs *Server) GetLightClientUpdatesByRange(w http.ResponseWriter, req *http.Request) {
+	// Prepare
+	ctx, span := trace.StartSpan(req.Context(), "beacon.GetLightClientUpdatesByRange")
+	defer span.End()
+
+	// Determine slots per period
+	config := params.BeaconConfig()
+	slotsPerPeriod := uint64(config.EpochsPerSyncCommitteePeriod) * uint64(config.SlotsPerEpoch)
+
+	// Adjust count based on configuration
+	countParam := req.URL.Query().Get("count")
+	count, err := strconv.ParseUint(countParam, 10, 64)
+	if err != nil {
+		http2.HandleError(w, fmt.Sprintf("got invalid 'count' query variable '%s', err %v", countParam, err),
+			http.StatusInternalServerError)
+		return
+	}
+
+	// Determine the start and end periods
+	startPeriodParam := req.URL.Query().Get("start_period")
+	startPeriod, err := strconv.ParseUint(startPeriodParam, 10, 64)
+	if err != nil {
+		http2.HandleError(w, fmt.Sprintf("got invalid 'start_period' query variable '%s', err %v", startPeriodParam, err),
+			http.StatusInternalServerError)
+		return
+	}
+
+	if count > config.MaxRequestLightClientUpdates {
+		count = config.MaxRequestLightClientUpdates
+	}
+	endPeriod := startPeriod + count - 1
+
+	// The end of start period must be later than Altair fork epoch, otherwise, can not get the sync committee votes
+	startPeriodEndSlot := (startPeriod+1)*slotsPerPeriod - 1
+	if startPeriodEndSlot < uint64(config.AltairForkEpoch)*uint64(config.SlotsPerEpoch) {
+		startPeriod = uint64(config.AltairForkEpoch) * uint64(config.SlotsPerEpoch) / slotsPerPeriod
+	}
+
+	headState, err := bs.HeadFetcher.HeadState(ctx)
+	if err != nil {
+		http2.HandleError(w, "could not get head state: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	lHeadSlot := uint64(headState.Slot())
```

Comment:
Can this be renamed to `headSlot`? We don't use such prefixes anywhere in the codebase. Similarly for other variables in this function

Reply:
Sure, patched.